api/include/opentelemetry/common/macros.h

```diff
@@ -90,6 +90,14 @@
 #  define OPENTELEMETRY_DEPRECATED_MESSAGE(msg)
 #endif
 
+// Regex support
+#if (__GNUC__ == 4 && (__GNUC_MINOR__ == 8 || __GNUC_MINOR__ == 9))
```

Comment:
nit: __GNUC_MINOR__ >= 8?

Reply:
The regex support is partially implemented for gcc 4.8 and gcc 4.9 only. So the existing condition looks good to me.

Reply:
But yeah, even  GNUC_MINOR >= 8 should work fine. But let it be explicit I think :)